nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -173,13 +173,18 @@ sub parseSystemdIni {
 #
 # If a directory with the same basename ending in .d exists next to the unit file, it will be
 # assumed to contain override files which will be parsed as well and handled properly.
-sub parseUnit {
-    my ($unitPath) = @_;
+sub parse_unit {
+    my ($unit_path) = @_;
 
     # Parse the main unit and all overrides
-    my %unitData;
-    parseSystemdIni(\%unitData, $_) for glob("${unitPath}{,.d/*.conf}");
-    return %unitData;
+    my %unit_data;
+    # Replace \ with \\ so glob() still works with units that have a \ in them
+    # Valid characters in unit names are ASCII letters, digits, ":", "-", "_", ".", and "\"
+    $unit_path =~ s/\\/\\\\/gmsx;
```

Comment:
```suggestion
    $unit_path =~ s/\\/\\\\/g;
```
Minor nitpick, but I don't believe the `msx` switches are needed here.

Reply:
Yes you are right, none of these are needed. But:
- https://metacpan.org/pod/Perl::Critic::Policy::RegularExpressions::RequireLineBoundaryMatching
- https://metacpan.org/pod/Perl::Critic::Policy::RegularExpressions::RequireDotMatchAnything
- https://metacpan.org/pod/Perl::Critic::Policy::RegularExpressions::RequireExtendedFormatting

Should I still remove them? I don't think they cause any issues when they are present

Reply:
Ah right, yeah keep them in then. Interesting reasoning, haven't been running `perlcritic` with low severity before :-)